### PR TITLE
Update URL parsing regexes and tests

### PR DESCRIPTION
**Pull request by adamjstewart**
@tgamblin this is what happens when I get triggered 😛 

### Before
```console
$ spack url summary
==> Generating a summary of URL parsing in Spack...

    Total URLs found:          3479
    Names correctly parsed:    3047/3479 (87.58%)
    Versions correctly parsed: 3161/3479 (90.86%)
```

### After
```console
$ spack url summary
==> Generating a summary of URL parsing in Spack...

    Total URLs found:          3465
    Names correctly parsed:    3193/3465 (92.15%)
    Versions correctly parsed: 3252/3465 (93.85%)
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/arm-forge/package.py [hunk 1]
```diff
@@ -19,107 +19,107 @@ class ArmForge(Package):
     #       versions (and checksums) based on the target platform shows up
 
     version(
-        "19.0.4-linux-rhel6-x86_64",
+        "19.0.4-Redhat-6.0-x86_64",
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/arm-forge/package.py]: This is kind of an unfortunate change in convention.

File: lib/spack/spack/test/url_parse.py [hunk 2]
```diff
@@ -109,24 +122,40 @@ def test_url_strip_version_suffixes(url, expected):
...
+    # Download version - complex
+    ('qt-everywhere-opensource-src-5.7.0', '5.7.0', 'qt'),
+    # Arch
+    ('VESTA-x86_64', '3.4.6', 'VESTA'),
     # VCS - bazaar
     ('libvterm-0+bzr681', '681', 'libvterm'),
     # License - gpl
-    ('PyQt-x11-gpl-4.11.3', '4.11.3', 'PyQt-x11')
+    ('PyQt-x11-gpl-4.11.3', '4.11.3', 'PyQt'),
+    ('PyQt4_gpl_x11-4.12.3', '4.12.3', 'PyQt4'),
```
[the remark this answers is not in the document]
Comment (adamjstewart) [reply lib/spack/spack/test/url_parse.py]: Does this even surprise you lol? You should see the `url_for_version` I wrote for `qt`.

File: var/spack/repos/builtin/packages/arm-forge/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/arm-forge/package.py]: I thought about it, and decided that when we add a `when=` parameter to the `version()` directive, all of these will become 19.0.4 anway, so it didn't really matter what we called the version. I can revert this package if you want, I don't really care.

File: var/spack/repos/builtin/packages/catch2/package.py [hunk 3]
```diff
@@ -6,11 +6,12 @@
 from spack import *
 
 
-class Catch(CMakePackage):
-    """Catch tests"""
+class Catch2(CMakePackage):
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/catch2/package.py]: But this package contains catch 1.* and catch 2.* sources :nerd_face: 
We could split the package, though.

Comment (ax3l) [reply var/spack/repos/builtin/packages/catch2/package.py]: Ok, consistent. Good with me then.

**Pull request merged**